packages/vue/src/components/organisms/SfHeader/_internal/SfHeaderNavigation.vue

```diff
@@ -20,10 +20,10 @@ export default {
   components: {
     SfSidebar,
   },
-  props: {
-    isVisibleOnMobile: {
-      type: Boolean,
-      default: false,
+  inject: ["mobileObserver"],
+  computed: {
```

Comment:
make this section of code into a mixin so that we can just reuse it in other components instead of copy/paste the same code several times.

Reply:
I'm not sure if we need that anywhere else as the mobile observer is imported normal way. Here I thought it can be imported in main component and than injected to intenral components, but maybe I should import it separatly for every component, wdyt?

Reply:
Afais, we need it in SfHeaderNavigation and the SfNavigationItem and SfTile. It should be imported separately instead of injecting from a main component, as we can't rely on user using SfHeaderNavigation for getting this observer. A very good example is SfTile :)